packages/context/src/context.ts

```diff
@@ -357,9 +363,9 @@ export class Context extends EventEmitter {
     this.registry.set(key, binding);
     if (existingBinding !== binding) {
       if (existingBinding != null) {
-        this.emit('unbind', existingBinding, this);
+        this.emit('unbind', existingBinding, this, 'unbind');
```

Comment:
Why do we have to repeat the event name in two places (the first and the last argument)? I find that cumbersome and making it easy to accidentally introduce mistakes in the future.

Reply:
Making the event type available in the `ContextEvent` object allows one function to listen on multiple event types to minimize overhead. In a strongly-typed language such as Java, the event class can be used as the discriminator.

Reply:
I see the point. Although JavaScript is not a strongly-typed language, right?

Have you considered adding a helper function?

```ts
// usage
this.emitBindingEvent('unbind', existingBinding);

// impl
emitBindingEvent(name: string, binding: Binding) {
  this.emit(name, binding, this, name);
}
```